lib/api/funnel.js

```diff
@@ -753,11 +753,29 @@ class Funnel {
    * @returns {Promise}
    */
   async executePluginRequest(request) {
-    try {
-      if (request.input.triggerEvents) {
-        const response = await this.processRequest(request);
-        return { ...response.result };
+    if (request.input.triggerEvents) {
+      try {
+        const { result } = await this.processRequest(request);
+        debug(
+          "Request %s successfully executed. Result: %a",
+          request.id,
+          result,
+        );
+        return global.kuzzle.pipe("request:afterExecution", {
+          request,
+          result,
+          success: true,
+        });
+      } catch (error) {
+        debug("Error processing request %s: %a", request.id, error);
+        return global.kuzzle.pipe("request:afterExecution", {
+          error: error,
+          request: request,
+          success: false,
```

Comment:
IMO this is the wrong events, maybe you should use the `request:onError` event instead. Since the request has not be executed in the right controller